EIPS/eip-3156.md

```diff
@@ -33,16 +33,40 @@ A flash lending feature integrates two smart contracts using a callback pattern.
 
 A `lender` MUST implement the IERC3156FlashLender interface.
 ```
+pragma solidity ^0.7.0 || ^0.8.0;
+import "./IERC3156FlashBorrower.sol";
+import "./IERC20.sol";
```

Comment:
This is a pretty major spec change.  Previously, this worked with *any* token type, the only requirement was that it was transferable and countable and referencable by address.  This change makes it so *only* ERC20 tokens are supported.  Is this intentional?

Also, I wonder if it would be worthwhile to enhance the specification to support stuff like ERC-1155 tokens as well?  Maybe address + extradata (byte array) and that extra data could be EIP-1155 token ID?

Another option would be to rename the functions to be like `flashLoanErc20(...)` and `flashFeeErc20(...)`.  Then another Flash Loan EIP could be created for EIP-1155 tokens that is very similar to this one and they are compatible so something could support either or both.

Reply:
I didn't think that by specifying IERC20 I would be barring other tokens from being lent, I'll revert that.

Support for ERC1155 tokens can be added as an extension to this ERC, same as we are doing for batch flash loans.